802-credential-sets.md

```diff
@@ -66,9 +86,79 @@ Similar tooling could choose to load the values by name from a database, vault,
 
 The created and modified timestamps are in [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) format.
 
-## Credential Injection
+## Parameter Mapping
+
+Parameter values can be mapped by the runtime using parameter sets, which works the same as credential sets.
+
+In the bundle.json parameters are defined:
+
+```json
+{
+ "parameters": {
+    "backend_port": {
+      "definition": "http_port",
+      "description": "The port that the back-end will listen on",
+      "destination": {
+        "env": "BACKEND_PORT"
+      }
+    }
+  }
+}
+```
 
-Credentials must be injected into the runtime of the invocation image. The following strategies are known to work:
+The following parameter set specifies that the hard-coded value "8080" should be used as the value for the backend_port parameter:
+
+```json
+{
+   "name": "local-dev-parameters",
+   "namespace": "dev",
+   "created": "2020-01-20T12:00:00.000Z",
+   "modified": "2020-01-28T18:30:00.111Z",
+   "labels": {
+     "env": "dev"
+   },
+   "parameters": [
+      {
+         "name": "backend_port",
+         "source": {
+            "value": "8080"
+         }
+      }
+   ]
+}
+```
+Source: [802.01-parameter-set.json](/examples/802.01-parameter-set.json)
+
+### Namespaces
+
+Credential and Parameter Sets MAY be scoped to a namespace.
```

Comment:
Question: Should we use RFC 2119 terminology for non-normative parts of the spec? It seems we're currently doing a bit of a mix but it might make sense to use it for parts that we envisage making normative.

(Higher up, I see we use "can" and "should" which is what triggered this question)

Reply:
I think it would be easier to maintain if we used the same language throughout the spec. Just so that as text is copied about you don't have to change it, and also so it's easier to spot when we mess up.

What do you think @technosophos?

Reply:
+1